Fix ShowFaPlusWindow applied on FA+ Game Mode

SimplyViper16 opened the pull request:
**Silver Fox strikes again!** He found another **buried** bug (idk how, he is like a hound) 🦊


**_Preconditions_**:
- The profile you are using have the _Display FA+ Window_ (_ShowFaPlusWindow_) mod **active**
- You start the game with the ITG GameMode but before do the first song, you change to FA+ GameMode

**_As Is_**:
During the Gameplay the Blue Fantastic with any type of judgement, never show up. It's always shown the White Fantastic.

I leave here some screenshots:
![ChangeModeITGToFAPlus](https://user-images.githubusercontent.com/10159143/174647540-e0bbe043-f9fb-4746-b576-7714263897c7.png)
![GameplayWhiteFantastic](https://user-images.githubusercontent.com/10159143/174647555-0129df77-c8f5-4bf0-ab45-9e017d534bc9.png)

_**What I've done**_:
Add check of the current GameMode along with the check of ShowFaPlusWindow

PS
I've also thought to call [ApplyMods()](https://github.com/Simply-Love/Simply-Love-SM5/blob/f362d57636e2a2f458184ce10cbeafa0bf70a1fc/Scripts/SL-PlayerOptions.lua#L798-L819) when change GameMode [here](https://github.com/Simply-Love/Simply-Love-SM5/blob/f362d57636e2a2f458184ce10cbeafa0bf70a1fc/BGAnimations/ScreenSelectMusic%20overlay/SortMenu/SortMenu_InputHandler.lua#L26-L40) but maybe can cause much more issues than benefits.

Viper 🐍

Commit:
Fix ShowFaPlusWindow applied on FA+ Game Mode

teejusb commented:
Thanks for pointing this out! This potentially might not be complete since the default assumption for `ShowFaPlusWindow` was that it would only be set in ITG mode, but clearly that's not the case as you have found out. What drawbacks do you think there might be to call `ApplyMods()`? It does sound like that it seems like it would be be okay and that function does loop to validate the options for the current mode.

Maybe @natano has some thoughts too?

Valentina16 commented:
> What drawbacks do you think there might be to call ApplyMods()?

I've seen `ApplyMods()` called only after the `LoadProfileCustom` and this comment [here](https://github.com/Simply-Love/Simply-Love-SM5/blob/f362d57636e2a2f458184ce10cbeafa0bf70a1fc/BGAnimations/ScreenProfileLoad%20overlay.lua#L8-L9) made me think that maybe is not safe to call that function in _"the middle of the code"_.
But now, after re-reading where the function is called, the only "problem" is that `SL[pn].ActiveModifiers` can be in _default state_ if you call `ApplyMods()` before `LoadProfileCustom` finished, which is never(?) the case when changing the game mode.

"What happens with the Guest profiles?" is the only open question I have.

teejusb commented:
We actually did just commit a PR that calls `ApplyMods()` for when players join/rejoin for fast profile switching in ITGmania (https://github.com/Simply-Love/Simply-Love-SM5/pull/399). So I think it should be fine. Briefly looking over the code for `ApplyMods()`, it more or less just looks at `SL[pn].ActiveModifiers` (or song state, etc.) to reference/update the modifiers. I think guest profiles should be fine as a result since the `ActiveModifiers` table would've already been "loaded" (i.e. set to a default state by the time we get to ScreenSelectMusic).

Commit:
Add ApplyMods for all human players when Change GameMode

Commit:
Add ApplyMods for all human players when Change GameMode

Fix ShowFaPlusWindow applied on FA+ Game Mode

Add ApplyMods for all human players when Change GameMode

Valentina16 commented:
@teejusb Did I make the change in the correct way? I mean, I've used the same PR to remove the initial commit and add the call to the function as we discussed.
Should I open a new PR with only the last commit?
Thank you!

natano commented:
I *think* this fix is correct. Maybe @jenxness has some input, because she had to deal with something like this when implementing quick profile switching.

natano commented:
I *think* this fix is correct. Maybe @jenxness has some input, because she had to deal with something like this when implementing quick profile switching.

jenxness commented:
I'm not familiar with what's actually going on theme- and engine-wise when the game mode changes so there are for sure things I'm not aware of, but my experience is that slapping `ApplyMods` on malfunctioning mod-related things usually helps. :laughing:

The biggest `ApplyMods` caveat that I had to deal with was that it was necessary to first reset the current state with `SL[pn]:initialize()` and (for GUESTs) [`GAMESTATE:ResetPlayerOptions`](https://itgmania.github.io/lua-for-itgmania/LuaAPI#Singletons-GameState-ResetPlayerOptions) before applying, otherwise you'd end up with an amalgamation of the mods of the original and the new profile, but that should not be necessary here since the profile doesn't change. :woman_shrugging:

natano reviewed (approved):
Looks good to me, but please squash the two commits into one.

SimplyViper16 committed:
Merge branch 'fix-faplus-judgement' of https://github.com/Valentina16/Simply-Love-SM5 into fix-faplus-judgement

Valentina16 commented:
VSCode drive me crazy today. Please be patient, I'll open new PR with the last clean commit. Sorry
